MapboxDirections/Match/MBMatchOptions.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+@objc(MBMatchingOptions)
+open class MatchingOptions: RouteOptions {
+    
+    public init(locations: [CLLocation], profileIdentifier: MBDirectionsProfileIdentifier? = nil) {
+        let waypoints = locations.map {
+            Waypoint(location: $0)
+        }
+        self.timestamps = locations.map { $0.timestamp }
+        super.init(waypoints: waypoints, profileIdentifier: profileIdentifier)
+    }
+    
+    @objc open var resampleTraces: Bool = false
+    
+    @objc private var timestamps: [Date]?
+    
+    public required init?(coder decoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override internal var params: [URLQueryItem] {
+        var params = super.params
+        
+        params.append(URLQueryItem(name: "tidy", value: String(describing: resampleTraces)))
+        
+        if let timestamps = timestamps, !timestamps.isEmpty {
+            let timeStrings = timestamps.map {
+                String(describing: $0.timeIntervalSince1970)
+                }.joined(separator: ";")
+            
+            params.append(URLQueryItem(name: "timestamps", value: timeStrings))
+        }
+        
+        return params
+    }
+    
+    override internal var path: String {
+        assert(!queries.isEmpty, "No query")
+        
+        let queryComponent = queries.joined(separator: ";")
+        return "matching/v5/\(profileIdentifier.rawValue)/\(queryComponent).json"
+    }
+    
+    internal func responseMatchOptions(from json: JSONDictionary) -> ([Tracepoint]?, [Match]?) {
+        var namedTracepoints: [Tracepoint]?
+        if let jsonWaypoints = (json["tracepoints"] as? [JSONDictionary]) {
+            namedTracepoints = zip(jsonWaypoints, self.waypoints).map { (api, local) -> Tracepoint in
+                let location = api["location"] as! [Double]
+                let coordinate = CLLocationCoordinate2D(geoJSON: location)
+                let alternateCount = api["alternatives_count"] as! Int
+                let waypointIndex = api["waypoint_index"] as! Int
+                let matchingIndex = api["matchings_index"] as! Int
+                let name = api["name"] as? String
+                // TODO: add name
+                // let possibleAPIName = api["name"] as? String
+                return Tracepoint(coordinate: coordinate, alternateCount: alternateCount, waypointIndex: waypointIndex, matchingIndex: matchingIndex, name: name)
+            }
+        }
+        
+        let tracePoints = namedTracepoints!
```

Comment:
@1ec5 On the `Waypoints` side of things, we fall back to [`self.waypoints`](https://github.com/mapbox/MapboxDirections.swift/blob/8ea5ad5f16f270c852985789156e302691cb5a35/MapboxDirections/MBRouteOptions.swift#L492) which doesn't make much sense here. 

Should we have a `tracePoints` property on `MatchingOptions` as well? This would introduce some complexity/refactoring I think.

Reply:
namedTracepoints shouldn’t be nil coming out the response, right? So we shouldn’t have to worry about falling back to self.waypoint, nor is a tracepoints property needed.